### PR TITLE
Change representation of PrivateKey to not use String

### PR DESCRIPTION
**Pull request by hunterpayne**
Changing the JwtSecretKey class to use an Array of Characters, byte array or PrivateKey.

### CODE REVIEW

**Comment by froth** [comment]
Thanks for your work. First glance looks fine to me, however I am currently on vacation and only have a phone with me ;) I can have another look at some point next week, sorry for the slow response. 

@gvolpe any thoughts? I think this is binary breaking as it changes the public interface of `JwtSecretKey` and in my opinion therefore should be released as 2.0.

File: core/src/main/scala/dev/profunktor/auth/jwt.scala
Comment (froth) [inline core/src/main/scala/dev/profunktor/auth/jwt.scala:35]: As we convert every type of key to `Array[Byte]` anyways: maybe it would be cleaner to model `JwtSecretKey` as `case class JwtSecretKey(key: Array[Char])` and do the conversions in the apply methods?

File: core/src/main/scala/dev/profunktor/auth/jwt.scala
Comment (froth) [inline core/src/main/scala/dev/profunktor/auth/jwt.scala]:
```suggestion
    @deprecated(message = "use of string to hold secret keys is deprecated and potentially unsafe", since = "2.x")
```

I think we have to publish this as 2.x, as it is already breaking and in 1.x there is no possibility to use anything but String

File: core/src/main/scala/dev/profunktor/auth/jwt.scala
Comment (froth) [inline core/src/main/scala/dev/profunktor/auth/jwt.scala]:
```suggestion
    @deprecated(message = "use of string to hold secret keys is deprecated and potentially unsafe", since = "2.x")
```

see above

File: core/src/main/scala/dev/profunktor/auth/jwt.scala
Comment (froth) [inline core/src/main/scala/dev/profunktor/auth/jwt.scala]: should this be commented in again?

**Review by froth — COMMENTED** [review]
Thanks for your work, some remarks/questions from my side. Again sorry for the delay.

**Comment by gvolpe** [comment]
@froth I leave it up to you guys, I don't use this library at all these days, don't wait for my review :)

File: core/src/main/scala/dev/profunktor/auth/jwt.scala
Comment (hunterpayne) [reply core/src/main/scala/dev/profunktor/auth/jwt.scala:35]: So some of the libs upon which you depend also have this same password as a String problem.  The way I wrote this PR, it delays making the password into a string as long as possible and avoiding that when the libs are fixed.  Pushing the conversion code into the apply method forces the conversions.  I'm trying to delay and avoid the conversions if possible.  Consider some new impl that just wants the raw bytes, forcing conversion early locks this out by design.

File: core/src/main/scala/dev/profunktor/auth/jwt.scala
Comment (hunterpayne) [reply core/src/main/scala/dev/profunktor/auth/jwt.scala]: No, this comment should be removed.

File: core/src/main/scala/dev/profunktor/auth/jwt.scala
Comment (froth) [reply core/src/main/scala/dev/profunktor/auth/jwt.scala:35]: That makes sense, thanks for explaining!

**Review by froth — APPROVED** [review]

File: core/src/main/scala/dev/profunktor/auth/jwt.scala
Comment (froth) [reply core/src/main/scala/dev/profunktor/auth/jwt.scala]:
```suggestion
        algorithms: Seq[JwtHmacAlgorithm]
```

**Comment by froth** [comment]
@hunterpayne thanks for your work. I like the changes and will try and merge them today. I will most likely have to change the base branch to series/2.0. I will try to change your pull request to reflect that, if I don't have the permissions to do so I will have to ask you to do that.

**Pull request merged**
